dev/benchmarks/macrobenchmarks/lib/src/color_filter_cache.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+import 'package:flutter/material.dart';
+
+class ColorFilterCachePage extends StatefulWidget {
+  const ColorFilterCachePage({Key? key}) : super(key: key);
+  @override
+  State<ColorFilterCachePage> createState() => _ColorFilterCachePageState();
+}
+
+class _ColorFilterCachePageState extends State<ColorFilterCachePage>
+    with TickerProviderStateMixin {
+  final ScrollController _controller = ScrollController();
+  @override
+  void initState() {
+    super.initState();
+    _controller.addListener(() {
+      if (_controller.offset < 50) {
+        _controller.animateTo(550, duration: const Duration(milliseconds: 1000), curve: Curves.ease);
+      } else if (_controller.offset > 500) {
+        _controller.animateTo(0, duration: const Duration(milliseconds: 1000), curve: Curves.ease);
+      }
+    });
+    Timer(const Duration(milliseconds: 1000), () {
+      _controller.animateTo(550, duration: const Duration(milliseconds: 1000), curve: Curves.ease);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.lightBlue,
+      body: ListView(
+        controller: _controller,
+        children: <Widget>[
+          const SizedBox(height: 500),
```

Comment:
We run these benchmarks on devices like the Moto G 4 which has a screen of around 360x640 (dpr adjusted) with the top bar taking up some of that vertical space. Have you run on such a device to see if the ColorFiltered has the visibility you want to measure on a screen of that size?

A gap of 500 before the ColorFiltered might mean that it will be barely visible when scrolled to 0. If you don't have access to such a device, I can do some testing myself to describe how it appears.